src/sentry/sentry_metrics/multiprocess.py

```diff
@@ -145,13 +154,31 @@ def poll(self) -> None:
         self.__next_step.poll()
 
         if self.__batch and self.__batch.ready():
-            self.__flush()
+            try:
+                self.__flush()
+            except MessageRejected:
```

Comment:
You call flush also in `join` which means you can receive a MessageRejected there as well, which can happen during rebalancing or when stopping the consumer. 
I think you will have to deal with that case as well. You can do it in a followup.